src/Trie.mo

```diff
@@ -389,8 +389,8 @@ module {
                 func (x : ?V, y : ?V) : V {
                   switch (x, y) {
                     case (null, null) { P.unreachable() };
-                    case (null, ?v) { v };
-                    case (?v, _) { v };
+                    case (?v, null) { v };
+                    case (_, ?v) { v };
```

Comment:
Documentation of `merge` says to prioritize the right trie. Either, we should adopt this fix, or change the documentation to say `merge` prioritizes the left trie.

Reply:
That's true. I will change documentation instead